lib/i18n/backend/active_record.rb

```diff
@@ -30,103 +27,111 @@ def initialize(*args)
         reload!
       end
 
-      def available_locales
-        Translation.available_locales
-      rescue ::ActiveRecord::StatementInvalid
-        []
-      end
+      module Implementation
+        include Base, Flatten
 
-      def store_translations(locale, data, options = {})
-        escape = options.fetch(:escape, true)
+        def available_locales
+          Translation.available_locales
+        rescue ::ActiveRecord::StatementInvalid
+          []
+        end
 
-        flatten_translations(locale, data, escape, false).each do |key, value|
-          translation = Translation.locale(locale).lookup(expand_keys(key))
+        def store_translations(locale, data, options = {})
+          escape = options.fetch(:escape, true)
 
-          if self.class.config.cleanup_with_destroy
-            translation.destroy_all
-          else
-            translation.delete_all
+          flatten_translations(locale, data, escape, false).each do |key, value|
+            translation = Translation.locale(locale).lookup(expand_keys(key))
+
+            if ActiveRecord.config.cleanup_with_destroy
+              translation.destroy_all
+            else
+              translation.delete_all
+            end
+
+            Translation.create(locale: locale.to_s, key: key.to_s, value: value)
           end
 
-          Translation.create(locale: locale.to_s, key: key.to_s, value: value)
+          reload! if ActiveRecord.config.cache_translations
         end
 
-        reload! if self.class.config.cache_translations
-      end
-
-      def reload!
-        @translations = nil
+        def reload!
+          @translations = nil
 
-        self
-      end
+          self
+        end
 
-      def initialized?
-        !@translations.nil?
-      end
+        def initialized?
+          !@translations.nil?
+        end
 
-      def init_translations
-        @translations = Translation.to_hash
-      end
+        def init_translations
+          @translations = Translation.to_hash
+        end
 
-      def translations(do_init: false)
-        init_translations if do_init || !initialized?
-        @translations ||= {}
-      end
+        def translations(do_init: false)
+          init_translations if do_init || !initialized?
+          @translations ||= {}
+        end
 
       protected
 
-      def lookup(locale, key, scope = [], options = {})
-        key = normalize_flat_keys(locale, key, scope, options[:separator])
-        key = key[1..-1] if key.first == '.'
-        key = key[0..-2] if key.last == '.'
+        def lookup(locale, key, scope = [], options = {})
+          key = normalize_flat_keys(locale, key, scope, options[:separator])
+          key = key[1..-1] if key.first == '.'
+          key = key[0..-2] if key.last == '.'
 
-        if self.class.config.cache_translations
-          keys = ([locale] + key.split(I18n::Backend::Flatten::FLATTEN_SEPARATOR)).map(&:to_sym)
+          if ActiveRecord.config.cache_translations
```

Comment:
There have been some changes since the `Implementation` module was removed(like this one for example)
So instead of reverting it back, let's wrap the existing implementation in the module and include it in the bottom, to make sure all changes that happened in between are there

Reply:
We have missed this at the time of rebase. Thanks for pointing it out 👍🏼